lib/epics/client.rb

```diff
@@ -183,12 +183,21 @@ def download(order_type, *args)
     document = order_type.new(self, *args)
     res = post(url, document.to_xml).body
     document.transaction_id = res.transaction_id
+    document.transaction_key = res.transaction_key
 
-    if res.segmented? && res.last_segment?
+    data = [res.order_data]
+
+    if res.segmented?
+      until res.last_segment?
+        document.segment_number = res.segment_number
+        res = post(url, document.to_fetch_segment_xml).body
+        res.transaction_key = document.transaction_key
```

Comment:
i think  this doesn't needs to be done for each segment. at least the fixtures don't transmit the transaction key and i assume thats why you put the rescue block around it